src/cobra/core/reaction.py

```diff
@@ -333,28 +495,29 @@ def flux(self):
             check_solver_status(self._model.solver.status)
             return self.forward_variable.primal - self.reverse_variable.primal
         except AttributeError:
-            raise RuntimeError("reaction '{}' is not part of a model".format(self.id))
+            raise RuntimeError(f"reaction '{self.id}' is not part of a model")
         # Due to below all-catch, which sucks, need to reraise these.
         except (RuntimeError, OptimizationError) as err:
-            raise_with_traceback(err)
+            raise err.with_traceback()
```

Comment:
More modern syntax:

```suggestion
            raise
```

one could also consider raising some kind of cobrapy error.

Reply:
I read [Raising exceptions](https://python-future.org/compatible_idioms.html#raising-exceptions) and it seems to indicate that Python 3 requires err.with_traceback()
Should I import traceback or something?

Reply:
Take a look at the official tutorial. I think the futures docs don't give the most recent impression since that is focused on compatible 2 & 3 syntax.

https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement

> If no expressions are present, [raise](https://docs.python.org/3/reference/simple_stmts.html#raise) re-raises the exception that is currently being handled, which is also known as the active exception. If there isn’t currently an active exception, a [RuntimeError](https://docs.python.org/3/library/exceptions.html#RuntimeError) exception is raised indicating that this is an error.